internal/proxy/meta_cache.go

```diff
@@ -560,6 +559,25 @@ func (m *MetaCache) GetShards(ctx context.Context, withCache bool, collectionNam
 
 	shards := resp.GetShards()
 
+	m.mu.Lock()
 	m.collInfo[collectionName].shardLeaders = shards
+	m.mu.Unlock()
+
 	return shards, nil
 }
+
+// ClearShards clear the shard leader cache of a collection
+func (m *MetaCache) ClearShards(collectionName string) {
+	log.Info("clearing shard cache for collection", zap.String("collectionName", collectionName))
+	m.mu.RLock()
+	_, ok := m.collInfo[collectionName]
+	m.mu.RUnlock()
+
+	if !ok {
+		return
+	}
+
+	m.mu.Lock()
+	m.collInfo[collectionName].shardLeaders = nil
+	m.mu.Unlock()
```

Comment:
Double check is needed if this method and `RemoveCollection` could be called simultaneously.

Reply:
ddl task is queued and never will be called concurrently